sound/soc/sof/intel/mtl.c

```diff
@@ -539,6 +539,8 @@ static irqreturn_t mtl_ipc_irq_thread(int irq, void *context)
 				snd_sof_ipc_get_reply(sdev);
 				snd_sof_ipc_reply(sdev, data->primary);
 
+				mtl_ipc_host_done(sdev);
```

Comment:
if the timing is critical,  how about put mtl_ipc_host_done(sdev) before snd_sof_ipc_reply(sdev, data->primary) which wakes up sleeping ipc msg and doesn't have any effect with FW side.